handler/src/main/java/io/netty/handler/ssl/OpenSslKeyMaterialManager.java

```diff
@@ -91,13 +93,14 @@ void setKeyMaterialClientSide(ReferenceCountedOpenSslEngine engine, String[] key
         }
     }
 
-    private void setKeyMaterial(ReferenceCountedOpenSslEngine engine, String alias) throws SSLException {
+    private boolean setKeyMaterial(ReferenceCountedOpenSslEngine engine, String alias) throws SSLException {
         OpenSslKeyMaterial keyMaterial = null;
         try {
             keyMaterial = provider.chooseKeyMaterial(engine.alloc, alias);
             if (keyMaterial != null) {
-                engine.setKeyMaterial(keyMaterial);
+                return engine.setKeyMaterial(keyMaterial);
             }
+            return true;
```

Comment:
Why true if key material is null?

Reply:
Got it, we only want to bail out if setting the material fails because the engine had been destroyed.

Reply:
it just signals that we want to continue... false should only be used when it was destroyed in the meantime

Reply:
Could simplify to `return keyMaterial == null || engine.setKeyMaterial(keyMaterial)`?